### PR TITLE
fix: catch errors thrown during enrichment

### PR DESCRIPTION
**Pull request by aaron-steinfeld**
## Description
Add a catch clause to prevent crashing the entire enricher on a failed enrichment. Instead, end enrichment for that trace (as later enrichers may have depended on the failed enrichment) and proceed.

### CODE REVIEW

**Review by skjindal93 — APPROVED** [review]

**Review by aaron-steinfeld — COMMENTED** [review]
Will update snyk and fix up some unnecessary breaking changes that were merged as part of https://github.com/hypertrace/hypertrace-ingester/pull/148 as well.

**Review by tim-mwangi — DISMISSED** [review]

**Review by tim-mwangi — APPROVED** [review]

**Comment by tim-mwangi** [comment]
Oops. Approved before build completed. I see it's failing

**Review by tim-mwangi — APPROVED** [review]

**Review by jcchavezs — APPROVED** [review]

**Comment by aaron-steinfeld** [comment]
> Oops. Approved before build completed. I see it's failing

snazzy new format checker, I love it!

File: hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/BackendEntityEnricher.java [hunk 1]
```diff
@@ -140,63 +139,51 @@ private void decorateWithBackendEntity(
...
-                                    BackendAttribute.BACKEND_ATTRIBUTE_PORT)))),
-                Pair.of(
-                    EntityConstants.getValue(BackendAttribute.BACKEND_ATTRIBUTE_PROTOCOL),
-                    createAttributeValue(
-                        backend
-                            .getAttributesMap()
-                            .get(
-                                EntityConstants.getValue(
-                                    BackendAttribute.BACKEND_ATTRIBUTE_PROTOCOL)))),
-                Pair.of(
```
Comment (aaron-steinfeld) [inline hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/BackendEntityEnricher.java:184]: This was being added twice, I removed it assuming a mistake.

File: hypertrace-trace-enricher/hypertrace-trace-enricher/build.gradle.kts [hunk 2]
```diff
@@ -46,10 +46,16 @@ dependencies {
...
+      because("https://snyk.io/vuln/SNYK-JAVA-IONETTY-1083991")
+    }
+    runtimeOnly("io.netty:netty-handler-proxy:4.1.60.Final") {
+      because("https://snyk.io/vuln/SNYK-JAVA-IONETTY-1083991")
+    }
   }
 
   // Required for the GRPC clients.
-  runtimeOnly("io.grpc:grpc-netty-shaded:1.35.0")
+  runtimeOnly("io.grpc:grpc-netty:1.35.0")
```
Comment (aaron-steinfeld) [inline hypertrace-trace-enricher/hypertrace-trace-enricher/build.gradle.kts:58]: also a bit of cleanup here - we don't shade netty in other deployments.

Comment (jcchavezs) [reply hypertrace-trace-enricher/hypertrace-trace-enricher/build.gradle.kts:58]: totally speaking out of ignorance (maybe unrelated here). Is there any action that tells us how bigger/smaller the artifact or docker image got after a change in PR? cc @JBAhire

Comment (aaron-steinfeld) [reply hypertrace-trace-enricher/hypertrace-trace-enricher/build.gradle.kts:58]: I suspect there's one out there, yeah (and if not, certainly seems feasible to build a trivial one). Was curious and just did a quick compare on the ht-trace-enricher image and there's a small improvement here: 
image (a89e306d4a74 is a build of current main, 4596e35d6c9c this branch)

**Pull request merged**